src/http/stream.ts

```diff
@@ -356,6 +355,20 @@ export class HttpStream extends Stream implements SqlOwner {
         });
     }
 
+    #fetchWithRetry(request: Request, retryCount = 3): Promise<Response> {
+        try {
+            const fetch = this.#fetch;
+            return fetch(request);
+        } catch (error: any) {
+            if (isRetryableError(error)) {
```

Comment:
Will that really solve the issue?
 Don't you need to in the catch of the promise here https://github.com/libsql/hrana-client-ts/blob/5d5bcdf73cc83f8b76f6bca4bcc56b6a5941a156/src/http/stream.ts#L350?

Unless fetch throws that error synchronously (I kinda doubt it does), I think it will escape your solution.

Reply:
@mateusfreira I think that point is too late to safely retry because it covers things like failing to process the response. For example, we may have successfully executed a `INSERT` statement and committed it to the database, but fail later on. Retrying would mean a duplicate insert.

Reply:
@mateusfreira I did switch to catching errors from the promise returned by `fetch()`, though, and added exponential backoff.